### PR TITLE
[PHP] Add includeDefaultValueFields to serializeToJsonString

### PR DESCRIPTION
**Pull request by yulia-0990**
Fix for #6035

### CODE REVIEW

**Comment by haberman** [comment]
Sorry for the slow review.

Can we use keyword arguments instead of positional arguments? eg. `serializeToJsonString(["foo" => true])`

Let's keep the `preserve_proto_fieldnames` option. It seems possible that people were using it even if it was unregistered. Even if it was not already being used, it should be available via keyword argument.

Let's call the new option `emit_defaults` to match Ruby: https://github.com/protocolbuffers/protobuf/blob/master/ruby/ext/google/protobuf_c/message.c#L1038-L1077

**Comment by yulia-0990** [comment]
Thank you for your review!

I hope I understood what you meant correctly. I made JSON options an array, also in C, and kept the `preserve_proto_fieldnames` option there (although it will not be compatible with its previous undocumented C implementation, as before it was a standalone boolean argument, not an array).

Unfortunately I could not use the new `Google\Protobuf\JsonSerializeOptions::EMIT_DEFAULTS` const in tests - it works for php, but for C it gives me `When using the C extension, we should not load runtime class` and I couldn't find how to fix it.

**Comment by haberman** [comment]
Sorry for the slow review.

It looks like you did not implement `preserve_proto_fieldnames` in pure PHP. Would you be able to do that (and add tests), so the two implementations respect the same options?

upb has also changed its API names, so the change will require a bit of renaming.

File: php/ext/google/protobuf/message.c [hunk 1]
```diff
@@ -764,6 +764,11 @@ PHP_METHOD(Message, mergeFromJsonString) {
   }
 }
 
+// JSON options /////////////////////////////////////////////////////////////////////
+
```
Comment (haberman) [inline php/ext/google/protobuf/message.c:768]: Please mention that this must match JsonSerializeOptions.php.

File: php/src/Google/Protobuf/JsonSerializeOptions.php [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+// DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+// THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+// (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+namespace Google\Protobuf;
+
+class JsonSerializeOptions
+{
+    const EMIT_DEFAULTS = "emit_defaults";
```
Comment (haberman) [inline php/src/Google/Protobuf/JsonSerializeOptions.php:37]: Add preserve_proto_fieldnames? Also please add a comment that this must match the C file.

**Comment by acozzette** [comment]
@yulia-0990 I am going to close this since there has been no activity in 30 days, but if you are still interested in working on this change then feel free to comment here and I can reopen it.

**Comment by s2x** [comment]
Hey @acozzette. Can I finish this task instead of @yulia-0990. I currently have a version build with PRESERVE_PROTO_FIELD_NAMES support. Additionally, I have written tests for both EMIT_DEFAULTS and PRESERVE_PROTO_FIELD_NAMES parameters.

My changes were created based on the current main branch. I just have a question, do I have to create a new pull request?

**Comment by acozzette** [comment]
Great! Yes, please create a new pull request.

**Pull request closed without merging**